### PR TITLE
feat: capture clicked elements on rageclicks

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

A customer asked if they could look at rageclicks by element. But we don't capture information about the rageclicked elements

Adds elements to rageclicks by injecting the autocapture capture method into the RageClick class instead of using the generic capture method

Screenshot 2023-01-18 at 14 01 26

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers

### CODE REVIEW

**Review by mariusandra — COMMENTED** [review]
I didn't run the code to verify, but on a quick read it seems hard to follow.

E.g. why did you remove passing of `instance` to RageClick? I also find it hard to understand calling `this._captureEvent` within `_captureEvent` through `rageclicks` through a passed `capture` callback. Is there any way to make this flow smoother?

Perhaps the existing code is just so bad and this was the easiest way. If so, I'm willing to step through it in a debugger and give it a proper review. 😅

File: src/autocapture.ts [hunk 1]
```diff
@@ -135,16 +135,18 @@ const autocapture = {
...
         if (isTextNode(target)) {
             // defeat Safari bug (see: http://www.quirksmode.org/js/events_properties.html)
             target = (target.parentNode || null) as Element | null
         }
 
-        if (e.type === 'click' && e instanceof MouseEvent) {
-            this.rageclicks?.click(e.clientX, e.clientY, new Date().getTime())
+        if (eventName === '$autocapture' && e.type === 'click' && e instanceof MouseEvent) {
+            this.rageclicks?.click(e.clientX, e.clientY, new Date().getTime(), (name: string) => {
+                this._captureEvent(e, instance, name)
```
[the remark this answers is not in the document]
Comment (pauldambra) [reply src/autocapture.ts]: Hmm, how about we separate the arrangement and the work.

rageClicks just decides if there has been a rageclick
autocapture emits the event if it does

**Review by mariusandra — APPROVED** [review]
Looks good to me now!
